### DISCUSSION TITLE
Scoring equation

### DISCUSSION THREAD

**Discussion by codepilot** [Xilinx/fpga24_routing_contest#37] [Q&A] [answered]
I think that I can calculate it, but I figured it would be good to ask the question to share it with the other contestants.

What is the ratio of wall seconds to critical route picoseconds?

I understand that the critical path is a more synthetic measurement than picoseconds, so it is rather approximate.

I am using vivado during development, so the critical route reported there is in picoseconds.

All things being equal, how many wall seconds should should be traded for how many picoseconds?

Daniel

**Comment by eddieh-xlnx** [comment]
> What is the ratio of wall seconds to critical route picoseconds?

Recently, we [updated](https://github.com/Xilinx/fpga24_routing_contest/pull/25) this repository with details on how to compute the critical-path *wirelength*, as well as reference tools on how to compute it. Specifically, the [scoring criteria](https://xilinx.github.io/fpga24_routing_contest/score.html#scoring-criteria) webpage states that saving nine wall clock seconds is equivalent to one unit saving of critical path wirelength. The intent of this disparity is to encourage contest routers to primarily focus on wall clock runtime to find a legal solution.

> I understand that the critical path is a more synthetic measurement than picoseconds, so it is rather approximate.

By "critical path" do you mean what we define as the "critical-path wirelength"? If so, your statement is correct. This metric is approximate when compared to the true sign-off quality critical-path delay (in picoseconds) that Vivado would return. This is intentional: we've designed this contest to not impose the heavy burden of having contestants build a fully timing-driven router (indeed, our differentiating focus is runtime). Instead, we've opted for a much more simplistic model that captures net delay only.

**Comment by codepilot** [comment]
Thanks, that helps. Does fan-out matter regarding timing?

Daniel

**Reply by eddieh-xlnx** [reply, marked as answer]
Fanout has a small effect in terms of *timing*, but it has no effect in terms of our definition of *wirelength*.
